p2p/state.py

```diff
@@ -52,12 +55,28 @@ def __init__(self,
         self.root_hash = root_hash
         self.scheduler = StateSync(root_hash, state_db)
         self._running_peers = set()  # type: Set[ETHPeer]
+        self._peers_with_pending_requests = {}  # type: Dict[ETHPeer, float]
         self.cancel_token = CancelToken('StateDownloader')
         if token is not None:
             self.cancel_token = self.cancel_token.chain(token)
 
     def register_peer(self, peer: BasePeer) -> None:
-        asyncio.ensure_future(self.handle_peer(cast(ETHPeer, peer)))
+        peer = cast(ETHPeer, peer)
+        asyncio.ensure_future(self.handle_peer(peer))
+        # A peer may get disconnected while we're waiting for a reply, so when it comes back we
+        # clear this.
```

Comment:
I don't understand this comment, specifically the "when it comes back".  I assume *it* is the peer, but what is it coming back from?

Reply:
Yes, _it_ is the peer, and it is coming back from the `PeerPool` -- this is the `.register_peer()` method, which is the callback used by `PeerPool` when it connects to a new peer. This is actually not needed, though, as when the `PeerPool` calls this we'll be passed a different `BasePeer` instance even if it's for a peer we were connected before. I thought it was needed because the `Node` class defines an `__eq__()` method that compares only the node's public ID, but that's not the case for `BasePeer`. I'm removing this, and it shouldn't be a problem that we're not removing old peers from that dict here because the `_periodically_retry_timedout()` coroutine does that